2636_implement-about-us-page

Cwagne17 opened the pull request:
This is the implementation of the about us page.  
- There are four components total
     1. Mission (top section)
     2. About-Clark (second section)
     3. Time line: in component is where an array stores json information for the timeline card (third section)
     4. Diversity Section (last section)

All the time line information is not in yet. I was going to wait to make any revisions before adding all of it as well as I need to know which photos everyone would like to go with certain events.  It is designed so a photo does not need to be present for each time line event.

ISSUE: In about-clark I tried implementing the usage statistics they are not appearing since I do not have a backend to clark set up and the usage stats have an if statement so that it only shows when data is there. So, I do not know if that looks how it is supposed to.


Desktop View:

![image](https://user-images.githubusercontent.com/70287489/99982164-3eed5a00-2d78-11eb-959f-b8e4df104c19.png)
![image](https://user-images.githubusercontent.com/70287489/99982240-5298c080-2d78-11eb-9047-036d3af1a5dc.png)
![image](https://user-images.githubusercontent.com/70287489/99982283-5d535580-2d78-11eb-83f5-ad5d1c6b33b1.png)
![image](https://user-images.githubusercontent.com/70287489/99982336-6ba17180-2d78-11eb-986a-9d0703c99a80.png)
![image](https://user-images.githubusercontent.com/70287489/99982367-73f9ac80-2d78-11eb-8a4d-2c5eafce805c.png)

Mobile View (iphone X):

![image](https://user-images.githubusercontent.com/70287489/99982466-94c20200-2d78-11eb-9ad4-36351583ff91.png)
![image](https://user-images.githubusercontent.com/70287489/99982493-a0152d80-2d78-11eb-9fd7-aa74d2175d95.png)
![image](https://user-images.githubusercontent.com/70287489/99982523-ac00ef80-2d78-11eb-976a-46c78fd0cc0c.png)
![image](https://user-images.githubusercontent.com/70287489/99982544-b4592a80-2d78-11eb-9648-a9a58a88e796.png)
![image](https://user-images.githubusercontent.com/70287489/99982565-bc18cf00-2d78-11eb-938b-51178e938621.png)
![image](https://user-images.githubusercontent.com/70287489/99982601-c6d36400-2d78-11eb-8a66-aa7b73179cfa.png)

Cwagne17 commented:
![AboutPageMobile](https://user-images.githubusercontent.com/70287489/100008217-204c8a80-2d9b-11eb-93a7-e678f89c87c8.png)
![AboutPageDesktop](https://user-images.githubusercontent.com/70287489/100008218-20e52100-2d9b-11eb-8b43-b0b54bbb7661.png)

I added json file for events. Added some of the timeline data. Changed some scss. and added usage component. I do not have a mission statement yet to put in as well as a full timeline with pictures that go along with some of the events.

EVogel1999 commented on src/app/cube/about-us/about-us.component.scss:
```diff
@@ -0,0 +1,7 @@
+@import '~_vars.scss';
+
+*{
```
Can we add 'background-color: white' so that way this is consistent with the theme we have for the details page?

Cwagne17 replied:
Sounds good will do that @EVogel1999

EVogel1999 commented on src/app/cube/about-us/components/mission/mission.component.html:
```diff
@@ -0,0 +1,10 @@
+  </div>
+  <div class="right-side">
+    <div class="text-container"> 
+      <h2>At Clark, we... the mission statement will go here in place of the words that are currently here. More stuff here</h2>
```
@pzalep1 Do you know if there is a mission statement written yet?  Otherwise, maybe about us should be a dark deploy.

pzalep1 replied:
I believe Dr. Kaza has one. We can ask at stand up.

EVogel1999 commented on src/app/cube/about-us/components/timeline/timeline.component.ts:
```diff
@@ -0,0 +1,17 @@
+})
+export class TimelineComponent implements OnInit {
+
+  events = events;
```
This looks like it should be deleted unless it is used in the timeline HTML file.

mfranz1 reviewed (commented):
Looks great so far...

Cwagne17 commented:
Just added new mission statement, white background, and added two new pictures for the timeline that are more appropriate.  All I need are some of the dates for the timeline as well as any other timeline events that want to be added. 

Only issue I am having, in events.json /* tslint:diable */ is there because I am not able to get the exclude in the tslint file to work.  However, ng serve will not run because comments aren't allowed in a json file. Any ideas? The error lint is giving me is " should be ' and semi colon at the end of the file. Both are not legal in a json.

pzalep1 commented on src/app/cube/about-us/components/timeline/events.json:
```diff
@@ -0,0 +1,36 @@
+      {
+          "id": 1,
+          "date": "Date",
+          "description": "684 Learning Objects 174 Users NSA NCCP NSF C5",
```
Can we add a small README in this module to explain how to add a new part of the timeline?

pzalep1 commented on tslint.json:
```diff
@@ -2,6 +2,11 @@
   ],
+  "linterOptions": {
+    "exclude": [
+        "events.json"
```
Instead of the disable-tslint in the json file try pointing to the file directly.

src/app/cube/about-us/components/timeline/events.json

pzalep1 commented on tsconfig.json:
```diff
@@ -9,7 +9,9 @@
     "declaration": false,
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
+    "esModuleInterop": true,
```
I think we can remove both of these if they did not solve the json issue.

pzalep1 reviewed (commented):
I left two comments in regards to the tslint and one small request for a README for the timeline.

pzalep1 commented:
Also, can we get updated screenshots with the white background? Thanks!

Cwagne17 commented:
![image](https://user-images.githubusercontent.com/70287489/100644878-e420a900-3309-11eb-93f6-133a2e45f8fa.png)
Here is the image with a white background.

Also the tsconfig.json changes allowed the component to read the json file. I will need to do another commit with those added back in.  

Here is what the read me looks like.
![image](https://user-images.githubusercontent.com/70287489/100645069-2cd86200-330a-11eb-81d1-8ba8059a2d0e.png)

EVogel1999 commented on src/assets/events.json:
```diff
@@ -0,0 +1,29 @@
+[
+      {
+          "date": "Date",
```
Why do some of these just say date?

pzalep1 replied:
These should match the dates on the timeline in the powerpoint.

Cwagne17 replied:
Some of the events had no dates just descriptions @EVogel1999 @pzalep1 
I will ask at standup for the dates

Cwagne17 commented:
I updated the timeline with Dr.Kaza as well as went through text changes with both Dr. Kaza, and Dr. Taylor.  
Here is a new screenshot. I believe it is ready to be published.

![image](https://user-images.githubusercontent.com/70287489/100907269-18c26b00-3498-11eb-96ac-4a2f3d1edf10.png)

EVogel1999 reviewed (approved):
Looks good.  Thank you!

pzalep1 commented:
One last question before we merge, is there a link in the footer for this page?

Cwagne17 commented:
@pzalep1 Just made the change, did you want to check to make sure I did it correctly?

pzalep1 commented on src/app/cube/shared/footer/footer.component.html:
```diff
@@ -16,8 +16,10 @@ <h3>Help</h3>
       </div>
       <div>
         <h3>General Information</h3>
+        <a aria-label= "Clickable Meet the Team link" href="http://clark.center/about" >{{copy.ABOUT}}</a>
```
This should be about us page for the aria-label.

pzalep1 reviewed (commented):
One last adjustment for the about us aria-label and then this can get versioned and merged.
